tests/unit/test_logging.py

```diff
@@ -75,3 +91,103 @@ def test_step_number(step: int) -> None:
             assert get_step_number() == 0
         assert get_step_number() == step
     assert get_step_number() == 0
+
+
+@unittest.mock.patch("trieste.models.gpflow.interface.tf.summary.scalar")
+def test_tensorboard_logging(mocked_summary_scalar: unittest.mock.MagicMock) -> None:
+    mocked_summary_writer = unittest.mock.MagicMock()
+    with tensorboard_writer(mocked_summary_writer):
+        data, models = {"A": mk_dataset([[0.0]], [[0.0]])}, {"A": _PseudoTrainableQuadratic()}
+        steps = 5
+        rule = FixedAcquisitionRule([[0.0]])
+        BayesianOptimizer(lambda x: {"A": Dataset(x, x ** 2)}, Box([-1], [1])).optimize(
+            steps, data, models, rule
+        )
+
+    ordered_scalar_names = [
+        "A.observation.best_overall",
+        "A.observation.best_new",
+        "Wallclock.total",
+        "Wallclock.query_point_generation",
+        "Wallclock.model_fitting",
+    ]
+    N = len(ordered_scalar_names)
+
+    for step in range(steps):  # iterate over saved scalars
```

Comment:
This is fine, if a bit confusing and error prone, plus it doesn't check that the mocked_summary_scalar.call_args_list doesn't contain superfluous entries. You should be able to simplify it something like:

```python
for call_arg, (step, scalar_name) in zip_longest(
    mocked_summary_scalar.call_args_list,
    product(range(steps), ordered_scalar_names
):
    assert call_arg[0][0] == scalar_name
    assert call_arg[-1]["step"] == step
    assert isinstance(call_arg[0][1], float)